packages/ckeditor5-table/tests/tablecaption/toggletablecaptioncommand.js

```diff
@@ -237,6 +276,22 @@ describe( 'ToggleTableCaptionCommand', () => {
 				'</table>'
 			);
 
+			// Show the caption.
+			command.execute();
+
+			setData( model,
+				'<table>' +
+					'<tableRow>' +
+						'<tableCell>' +
+							'<paragraph></paragraph>' +
+						'</tableCell>' +
+					'</tableRow>' +
+					'<caption></caption>' +
+				'</table>'
+			);
```

Comment:
This would override the model and the original issue would not happen. You should modify the model by removing the caption content.